MPSGameLift/Code/Source/MatchmakingSystemComponent.cpp

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project. For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ * 
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "MatchmakingSystemComponent.h"
+
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/RTTI/BehaviorContext.h>
+
+#include <AWSCoreBus.h>
+#include <ResourceMapping/AWSResourceMappingBus.h>
+
+#include <Framework/ServiceRequestJob.h>
+
+
+namespace MPSGameLift
+{
+    namespace ServiceAPI
+    {
+        struct PlayerSkill
+        {
+            bool OnJsonKey(const char* key, AWSCore::JsonReader& reader)
+            {
+                if (strcmp(key, "N") == 0)
```

Comment:
General comment for these messages. Do we or could we have some docs on this? Just looking at the comparisons here it's not clear what is going on. Now about adding samples of the incoming or outgoing json messages and what keys and values are being looked for in these structs.

Reply:
Thank you! Added comments and adding doc urls:
https://docs.aws.amazon.com/gamelift/latest/apireference/API_MatchmakingTicket.html
https://docs.aws.amazon.com/gamelift/latest/apireference/API_Player.html